File: Tutorials/ASP.NET/MVC5/CS/Controllers/BaseController.cs
using System;
using System.Web.Mvc;
using DevExpress.Xpo;

namespace AspNetMvcApplication.Controllers {

    public abstract class BaseController : Controller {
        protected readonly UnitOfWork UnitOfWork;
        public BaseController() {
            UnitOfWork = new UnitOfWork(XpoDefault.DataLayer);
        }
        public void SafeExecute(Action method) {
            try {
                method();
            } catch(Exception e) {
                ViewData["EditError"] = e.Message;
            }
        }
        protected override void Dispose(bool disposing) {
            base.Dispose(disposing);
            if(disposing) {
                UnitOfWork.Dispose();
            }
        }
    }
}

File: Tutorials/ASP.NET/MVC5/VB/Controllers/BaseController.vb
Imports System
Imports System.Web.Mvc
Imports DevExpress.Xpo

Namespace AspNetMvcApplication.Controllers

    Public MustInherit Class BaseController
        Inherits Controller

        Protected ReadOnly UnitOfWork As UnitOfWork
        Public Sub New()
            UnitOfWork = New UnitOfWork(XpoDefault.DataLayer)
        End Sub
        Public Sub SafeExecute(ByVal method As Action)
            Try
                method()
            Catch e As Exception
                ViewData("EditError") = e.Message
            End Try
        End Sub
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            MyBase.Dispose(disposing)
            If disposing Then
                UnitOfWork.Dispose()
            End If
        End Sub
    End Class
End Namespace